### PR TITLE
UI: Save screenshot of main output to PNG

### PR DESCRIPTION
**Pull request by zavitax**
Add screenshot settings section
Add screenshot output directory setting
Add save output screenshot hotkey binding
Add swscale swresample libs to UI CMakeLists.txt
Use existing ffmpeg bindings to save to PNG

### CODE REVIEW

File: UI/platform-osx.mm [hunk 1]
```diff
@@ -84,6 +84,21 @@ string GetDefaultVideoSavePath()
 	return url.path.fileSystemRepresentation;
 }
 
+string GetDefaultScreenShotsSavePath()
+{
+	NSFileManager *fm = [NSFileManager defaultManager];
+	NSURL *url = [fm URLForDirectory:NSMoviesDirectory // TODO: Change this to default pictures directory
```
Comment (palana) [inline UI/platform-osx.mm]: `NSPicturesDirectory` re: your TODO

File: UI/window-basic-main.cpp [hunk 2]
```diff
@@ -1596,6 +1612,207 @@ void OBSBasic::DeferredLoad(const QString &file, int requeueCount)
...
+	// Save screenshot to specified file path
+	bool success = SaveScreenshotToFile(strOutputFilePath.c_str());
+
+	if (success)
+	{
+		// TODO: Audio/visual feedback
+	}
+}
+
+bool OBSBasic::SaveScreenshotToFile(const char* outputFilePath)
```
Comment (palana) [inline UI/window-basic-main.cpp]: This function is … really big; you should consider splitting it up, and maybe even moving (all) the parts to a separate file (e.g. not have any ffmpeg includes in UI/window-basic-main.cpp)

File: UI/window-basic-settings.cpp [hunk 3]
```diff
@@ -270,6 +270,7 @@ void OBSBasicSettings::HookWidget(QWidget *widget, const char *signal,
 #define VIDEO_RESTART   SLOT(VideoChangedRestart())
 #define VIDEO_RES       SLOT(VideoChangedResolution())
 #define VIDEO_CHANGED   SLOT(VideoChanged())
+#define SCREENSHOTS_CHANGED   SLOT(ScreenShotsChanged())
```
Comment (palana) [inline UI/window-basic-settings.cpp]: Super-nitpick: just use a single space here?

File: UI/forms/OBSBasicSettings.ui [hunk 4]
```diff
@@ -3735,6 +3744,59 @@
...
+           </property>
+           <property name="alignment">
+            <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
+           </property>
+           <property name="buddy">
+            <cstring>screenshotOutputPath</cstring>
+           </property>
+          </widget>
+         </item>
+                <item row="0" column="1">
```
Comment (palana) [inline UI/forms/OBSBasicSettings.ui]: The formatting below (and including) this line is kind of weird; reformat it to better match the other lines?

**Comment by palana** [comment]
One other thing: pull requests should never* contain merge commits, please remove https://github.com/obsproject/obs-studio/pull/1268/commits/ab173c0fc6a87c034a9aefb3104c042c72e26a82

**Comment by zavitax** [comment]
Closing this PR in favor of the revised https://github.com/obsproject/obs-studio/pull/1269

**Pull request closed without merging**